tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -34,6 +34,28 @@ sub get_script_run {
     return $pre_test_cmd;
 }
 
+sub analyzeResult {
+    my ($self, $text) = @_;
+    my $result;
+    $text =~ /Overall guest upgrade result is:(.*)Test done/s;
+    my $rough_result = $1;
+
+    foreach (split("\n", $rough_result)) {
+        if ($_ =~ /(\S+)\s+\-{3}\s+(PASS|FAIL|SKIP|TIMEOUT)(\s+\-{3}\s+(.*))*/ig) {
+            my ($testcase_name, $status, $error) = ($1, $2, $4);
```

Comment:
Just a small one, we could use named regexp groups. 
`/(?<testcase_name>\S+)\s+\-{3}\s+(PASS|FAIL|SKIP|TIMEOUT)(\s+\-{3}\s+(.*))*` and then you can access it with  `$+{testcase_name}`. Be aware that this hash will get overwritten after you call other match, so you still need variables assignment.